### DISCUSSION TITLE
Failing to login to apiman while running in the localhost

### DISCUSSION THREAD

**Discussion by winstondavid829** [apiman/apiman#2430] [Help]
Hi, I have setup the apiman as mentioned in the README.adoc file and changing the API_HOSTNAME to localhost. While trying to login through the portal using the login portal. The application is throwing Forbidden error and in console it is showing below given error,

apiman-manager_1    | 03:54:03,432 ERROR [org.keycloak.adapters.OAuthRequestAuthenticator] (default task-1) failed to turn code into token
apiman-manager_1    | 03:54:03,432 ERROR [org.keycloak.adapters.OAuthRequestAuthenticator] (default task-1) status from server: 401
apiman-manager_1    | 03:54:03,432 ERROR [org.keycloak.adapters.OAuthRequestAuthenticator] (default task-1)    {"error":"unauthorized_client","error_description":"Invalid client secret"}

What am I missing in the registration of the application. Can someone help me to figure it out please.

**Comment by msavy** [comment]
Does it work if you change it back to `local.gd`? Are you saying it only happens if you change to `localhost`.

**Reply by winstondavid829** [reply]
Nope. When I change it back to local.gd the path cannot be identified with a 404 error. but when the path is at localhost it is working and while trying to login this error occurs. 

I installed the version using docker compose. It will be helpful if you could help me on installing it.

**Reply by msavy** [reply]
Sounds like your secrets are out of sync. Maybe you ran setup multiple times, which does not work. 

```
Destroy all of the Docker resources:
 
docker compose -f docker-compose.setup.yml down -v
docker compose down -v
```

Delete your `apiman-docker-compose` directory and extract again freshly (i.e. ensure you got rid of all generated data such as secrets and certificates).

Start instructions from scratch again.

**Reply by msavy** [reply]
Are you up and running again? All good?

**Reply by ASHISHTOM** [reply]
I have been trying docker-compose version. I am trying this in GCP VM. But when using `local.gd` as `APIMAN_HOSTNAME` , I am not able to access it outside. It is giving 404 error, but no errors observed in logs. 

I tried to give public IP of my GCP VM as `APIMAN_HOSTNAME` , but it is not working. 
`http://gateway.${APIMAN_HOSTNAME}:${HTTP_PORT}` URLs like this becomes `http://gateway.34.4.4.4:8080` which makes no sense, since there is no DNS configured.

What all changes needed to change local.gd to smthg else like a public IP? @msavy

**Reply by ghost** [reply]
@ASHISHTOM Were you able to resolve it?

**Reply by msavy** [reply]
Have you tried using nip.io instead? Might be more helpful if you are trying to use an IP address
